### ISSUE TITLE
Agent state "Not ready" on Ubuntu 18.04 after Veeam restore

### ISSUE THREAD

**Issue by ekrof** [Azure/WALinuxAgent#1894]
In the process of migrating our data center to Azure Stack, the VMs running Ubuntu 18.04 bionic gets stuck in the agent state 'Not Ready' after restoring from Veeam. This is regardless of kernel version. 

```
Ubuntu 18.04.04 LTS
WALinuxAgent-2.2.45
Python: 3.6.9
Goal state agent: 2.2.45

Azure Stack 1907
```

**What I've tried**

- Reboot
- Updating kernel (Both Azure specific (5.3.0-1020-azure and latest)
- Reinstalling the walinuxagent
- Building walinuxagent from source
- Opening all WireServer ports
- Turning off and on DHCP
- Switching VNETs
- Using Proxy VMs in Veeam restore menu
- Tried connecting via jumphost

Boot diagnostics say nothing of value. Because of this I cannot SSH into the VM so I'm pretty much locked outside. VMs running Ubuntu 14.04 and 16.04 works fine and restores normally.

**Comment by narrieta** [comment]
@byue - could you take a look at this issue? thanks!

**Comment by byue** [comment]
Sure, np. @ekrof when you migrated VMs, did you use "attach" semantics when attaching OS disk, instead of normal "FromImage" semantic? "Attach" semantics will let control plane know that the VM is fully provisioned, if "FromImage" was used then control plane will get stuck with Not Ready signal. The stamp version is on 1907 and is unsupported, recommendation is to move to 2002.

**Comment by ekrof** [comment]
HI @byue thanks for the response. I can't recall anything that had to do with semantics when restoring back then. Since I opened the issue I did manage to restore the VMs some other way so It's all good now. 1907 was supported back in May 2020, so rest assured we have updated our AzS stamp v since then 😄

I'll be closing the issue and if anyone else is stuck on the same problem in the future, follow @byue's recommendation of using Attach semantics.
